internal/controller/adapters/sveltos/serviceset_controller_test.go

```diff
@@ -15,9 +15,354 @@
+	Expect(err).NotTo(HaveOccurred())
+	Expect(cl).NotTo(BeNil())
+})
+
+var _ = Describe("ServiceSet Controller Integration Tests", func() {
+	const (
+		testClusterName = "test-cluster"
+		testTemplate    = "test-template"
+		testServiceName = "test-service"
+		testNamespace   = "test-namespace"
+		testCredential  = "test-credential"
+	)
+
+	testTimeout := &metav1.Duration{Duration: time.Minute * 5}
+
+	type tc struct {
+		name string
+		src  *kcmv1.ServiceHelmOptions
+		dst  *kcmv1.ServiceHelmOptions
+		want *kcmv1.ServiceHelmOptions
+	}
+
+	DescribeTable("merge behavior",
+		func(t tc) {
+			mergeHelmOptions(t.src, t.dst)
+			Expect(reflect.DeepEqual(t.dst, t.want)).To(BeTrue(),
+				"test case failed: %s\ndst=%#v\nwant=%#v", t.name, t.dst, t.want)
+		},
+
+		Entry("src=nil → no change",
+			tc{
+				name: "src nil",
+				src:  nil,
+				dst:  &kcmv1.ServiceHelmOptions{},
+				want: &kcmv1.ServiceHelmOptions{},
+			},
+		),
+
+		Entry("dst=nil → safely does nothing",
+			tc{
+				name: "dst nil",
+				src:  &kcmv1.ServiceHelmOptions{Atomic: ptr.To(true)},
+				dst:  nil,
+				want: nil,
+			},
+		),
+
+		Entry("src empty → dst unchanged",
+			tc{
+				name: "src empty",
+				src:  &kcmv1.ServiceHelmOptions{},
+				dst:  &kcmv1.ServiceHelmOptions{Atomic: ptr.To(false)},
+				want: &kcmv1.ServiceHelmOptions{Atomic: ptr.To(false)},
+			},
+		),
+
+		Entry("copy all boolean fields",
+			tc{
+				name: "copy all bools",
+				src: &kcmv1.ServiceHelmOptions{
+					EnableClientCache:        ptr.To(true),
+					DependencyUpdate:         ptr.To(true),
+					Wait:                     ptr.To(false),
+					WaitForJobs:              ptr.To(true),
+					CreateNamespace:          ptr.To(false),
+					SkipCRDs:                 ptr.To(true),
+					Atomic:                   ptr.To(false),
+					DisableHooks:             ptr.To(true),
+					DisableOpenAPIValidation: ptr.To(true),
+					SkipSchemaValidation:     ptr.To(true),
+					Replace:                  ptr.To(false),
+				},
+				dst: &kcmv1.ServiceHelmOptions{},
+				want: &kcmv1.ServiceHelmOptions{
+					EnableClientCache:        ptr.To(true),
+					DependencyUpdate:         ptr.To(true),
+					Wait:                     ptr.To(false),
+					WaitForJobs:              ptr.To(true),
+					CreateNamespace:          ptr.To(false),
+					SkipCRDs:                 ptr.To(true),
+					Atomic:                   ptr.To(false),
+					DisableHooks:             ptr.To(true),
+					DisableOpenAPIValidation: ptr.To(true),
+					SkipSchemaValidation:     ptr.To(true),
+					Replace:                  ptr.To(false),
+				},
+			},
+		),
+
+		Entry("copy Timeout",
+			tc{
+				name: "copy timeout",
+				src:  &kcmv1.ServiceHelmOptions{Timeout: testTimeout},
+				dst:  &kcmv1.ServiceHelmOptions{},
+				want: &kcmv1.ServiceHelmOptions{Timeout: testTimeout},
+			},
+		),
+
+		Entry("copy map pointer",
+			tc{
+				name: "copy labels map",
+				src:  &kcmv1.ServiceHelmOptions{Labels: &map[string]string{"env": "prod"}},
```

Comment:
Question: when labels are provided in both `src` and `dst`, should they be merged or overridden? If they should be merged, I can suggest adding a test case for it (the current test entry verifies that labels are overwritten).

Reply:
good point added that